solr/modules/opentelemetry/src/java/org/apache/solr/opentelemetry/ClosableTracerShim.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.solr.opentelemetry;
+
+import io.opentelemetry.sdk.trace.SdkTracerProvider;
+import io.opentracing.Scope;
+import io.opentracing.ScopeManager;
+import io.opentracing.Span;
+import io.opentracing.SpanContext;
+import io.opentracing.Tracer;
+import io.opentracing.propagation.Format;
+import java.lang.invoke.MethodHandles;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Delegate shim that forwards all calls to the actual {@link
+ * io.opentelemetry.opentracingshim.OpenTracingShim}, and in addition calls {@link
+ * SdkTracerProvider#close()} to really close the OTEL SDK tracer when the OT shim is closed.
+ *
+ * <p>TODO: This can be removed once we migrate Solr instrumentation from OpenTracing to
+ * OpenTelemetry
+ */
+public class ClosableTracerShim implements Tracer {
```

Comment:
I'd have subclassed [TracerShim](https://github.com/open-telemetry/opentelemetry-java/blob/v1.21.0/opentracing-shim/src/main/java/io/opentelemetry/opentracingshim/TracerShim.java#L20) instead if it was not final.